components/org.wso2.carbon.identity.api.server.organization.user.sharing.management/org.wso2.carbon.identity.api.server.organization.user.sharing.management.v1/src/main/java/org/wso2/carbon/identity/api/server/organization/user/sharing/management/v1/core/UsersApiServiceCore.java

```diff
@@ -76,21 +84,187 @@ public UsersApiServiceCore(UserSharingPolicyHandlerService userSharingPolicyHand
      *
      * @param userShareRequestBody Contains details for user sharing.
      */
-    public void shareUser(UserShareRequestBody userShareRequestBody) throws UserSharingMgtException {
+    public Response shareUser(UserShareRequestBody userShareRequestBody) {
 
         if (userShareRequestBody == null) {
-            throw new UserSharingMgtClientException(INVALID_SELECTIVE_USER_SHARE_REQUEST_BODY.getCode(),
-                    INVALID_SELECTIVE_USER_SHARE_REQUEST_BODY.getMessage(),
-                    INVALID_SELECTIVE_USER_SHARE_REQUEST_BODY.getDescription());
+            return Response.status(Response.Status.BAD_REQUEST)
+                    .entity(buildErrorResponse(makeRequestError(INVALID_SELECTIVE_USER_SHARE_REQUEST_BODY))).build();
         }
 
         // Populate selectiveUserShareDO object from the request body.
+        SelectiveUserShareDO selectiveUserShareDO = populateGeneralUserShareDO(userShareRequestBody);
```

Comment:
is `populateGeneralUserShareDO()` is used to populate a `selectiveUserShareDO` as well?